rust_src/src/process.rs

```diff
@@ -102,6 +116,102 @@ pub fn set_process_plist(process: LispObject, plist: LispObject) -> LispObject {
     }
 }
 
+/// Return the status of PROCESS.
+/// The returned value is one of the following symbols:
+/// run  -- for a process that is running.
+/// stop -- for a process stopped but continuable.
+/// exit -- for a process that has exited.
+/// signal -- for a process that has got a fatal signal.
+/// open -- for a network stream connection that is open.
+/// listen -- for a network stream server that is listening.
+/// closed -- for a network stream connection that is closed.
+/// connect -- when waiting for a non-blocking connection to complete.
+/// failed -- when a non-blocking connection has failed.
+/// nil -- if arg is a process name and no such process exists.
+/// PROCESS may be a process, a buffer, the name of a process, or
+/// nil, indicating the current buffer's process.
+#[lisp_fn]
+pub fn process_status(process: LispObject) -> LispObject {
+    let mut p = process;
```

Comment:
Could this be:

    let p = if process.is_string() {
        get_process(process)
    } else {
       LispObject::from( unsafe {  cget_process(process.to_raw()) } )
    }